packages/artifact/src/internal/upload-gzip.ts

```diff
@@ -0,0 +1,51 @@
+import * as fs from 'fs'
+import * as zlib from 'zlib'
+
+/**
+ * Creates a Gzip compressed file of an original file at the provided temporary filepath location
+ * @param {string} originalFilePath filepath of whatever will be compressed. The original file will be unmodified
+ * @param {string} tempFilePath the location of where the Gzip file will be created
+ * @returns the size of gzip file that gets created
+ */
+export async function createGZipFileOnDisk(
+  originalFilePath: string,
+  tempFilePath: string
+): Promise<number> {
+  return new Promise((resolve, reject) => {
+    const inputStream = fs.createReadStream(originalFilePath)
+    const gzip = zlib.createGzip()
+    const outputStream = fs.createWriteStream(tempFilePath)
+    inputStream.pipe(gzip).pipe(outputStream)
+    outputStream.on('finish', () => {
+      // wait for stream to finish before calculating the size which is needed as part of the Content-Length header when starting an upload
+      const size = fs.statSync(tempFilePath).size
```

Comment:
I am guessing those `sync` functions are slowing down your throughput.  Can you `await` on the async version?

Reply:
Changed it up using `promisify` and the async `fs.stat`

Reply:
This is still sync, did you upload the latest code yet?